enigma-swing/src/main/java/cuchaz/enigma/gui/panels/StructurePanel.java

```diff
@@ -40,16 +40,19 @@ public StructurePanel(Gui gui) {
 
         this.optionsPanel.add(this.obfuscationVisibilityLabel, cb.pos(0, 0).build());
         this.obfuscationVisibility = new JComboBox<>(StructureTreeOptions.ObfuscationVisibility.values());
+        this.obfuscationVisibility.setRenderer((list, value, index, isSelected, cellHasFocus) -> new JLabel(I18n.translate("structure.options.obfuscation." + value.getKey())));
```

Comment:
Unfortunately it's not as easy as that, this:

- doesn't highlight the selected element
- will create a new JLabel every time the combobox asks to rerender the label

You can however extend DefaultListCellRenderer which does all that for you, override getListCellRendererComponent to calls super (that method does the selection highlighting among other stuff), and then sets the label text. A lot of classes extend DefaultListCellRenderer so that shouldn't be a problem.

Also at this point I'd have the full localization key in the enum as opposed to concating it every time.

Reply:
Done!